shell/platform/embedder/embedder.h

```diff
@@ -1010,6 +1061,9 @@ FlutterEngineResult FlutterEngineCreateAOTData(
 FLUTTER_EXPORT
 FlutterEngineResult FlutterEngineCollectAOTData(FlutterEngineAOTData data);
 
+/// The order, type, and size of these struct members must remain the same, and
+/// members should not be removed. This is to allow for forward and backward
+/// compatibility between the engine and the embedder.
```

Comment:
It seems odd, and noisy, to put this boilerplate everywhere. It's also odd to explicitly call this out N times, when don't mention that function names can't change, argument numbers/orders/types can't change, core behavior of functions can't change, etc. I'm not sure why this particular piece of ABI preservation needs to be called out specially.

A file-level comment that says that this header presents a stable ABI, and gives examples like function and struct changes, seems like a better option if we want to comment thing like this.

@chinmaygarde for thoughts.